### ISSUE TITLE
Pod disappears after creation

### ISSUE THREAD

**Issue by aivanou** [volcano-sh/volcano#1688]
**What happened**:
I have a simple pod that just produces python RuntimeException. When I try to run it with the following yaml:

``` yaml
apiVersion: batch.volcano.sh/v1alpha1
kind: Job
metadata:
  name: sh-fb9dd3d3
spec:
  maxRetry: 0
  plugins:
    env: []
    svc: []
  queue: test
  schedulerName: volcano
  tasks:
  - maxRetry: 0
    name: sh-0
    policies:
    - action: RestartJob
      event: PodEvicted
    - action: RestartJob
      event: PodFailed
    replicas: 1
    template:
      spec:
        containers:
        - command:
          - /bin/sh
          - -c
          - python my_job.py
          env: []
          image: public.ecr.aws/y5x3w0a7/aivanou-tests
          name: sh-0
          ports: []
          resources:
            limits: {}
            requests: {}
        restartPolicy: Never
```

The pod starts but then it completely disappears from pods list: `kubectl get pods -n default`

The pod python code:
```python
raise RuntimeError('test error')
```

It seems that the pod starts running, fails with error and then getting restarted. After restart it fails and never shows up in the kubectl pods. 

The output of `vcclt job --name  sh-fb9dd3d3`

```
Name:       	sh-fb9dd3d3
Namespace:  	default
Labels:     	<none>
Annotations:	<none>
API Version:
Kind:
Metadata:
  Creation Timestamp:	2021-08-20 00:06:22 +0000 UTC
  Generate Name:
  Generation:        	1
  Resource Version:  	2148088
  Self Link:
  UID:               	3fdbc1e9-3afe-401f-bdd1-685aeefe6d4a
Spec:
  Min Available:     	1
  Plugins:
    Env:	[]
    Ssh:	[]
  Scheduler Name:    	volcano
  Tasks:
    Name:	sh-0
    Replicas:	1
    Template:
      Metadata:
        Annotations:
          Cri . Cci . Io / Container - Type:
          Kubernetes . Io / Availablezone:
          Network . Alpha . Kubernetes . Io / Network:
        Creation Timestamp:	0001-01-01 00:00:00 +0000 UTC
      Spec:
        Containers:
          Command:
            /bin/sh
            -c
            python my_job.py
          Image:	public.ecr.aws/y5x3w0a7/aivanou-tests
          Name: 	sh-0
          Ports:
          Resources:
            Limits:
              Cpu:   	0
              Memory:	0
            Requests:
              Cpu:   	0
              Memory:	0
            Working Dir:
        Image Pull Secrets:
        Restart Policy:   	Never
Status:
  RetryCount:   	3
  Min Available:	1
  Version:      	7
  State:
    Phase:	Failed
Events:
Type           	Reason                                  	Age                           	Form                                    	Message
-------        	-------                                 	-------                       	-------                                 	-------
Warning        	Unschedulable                           	15m (x2 over 15m)             	volcano                                 	0/0 tasks in gang unschedulable: pod group is not ready, 1 minAvailable.
Warning        	PodGroupPending                         	15m (x18 over 15m)            	vc-controller-manager                   	PodGroup default:sh-fb9dd3d3 unschedule,reason: 1/0 tasks in gang unschedulable: pod group is not ready, 1 minAvailable.
Normal         	Scheduled                               	15m (x3 over 15m)             	volcano                                 	pod group is ready
Normal         	ExecuteAction                           	15m (x3 over 15m)             	vc-controller-manager                   	Start to execute action RestartJob
Warning        	Unschedulable                           	15m (x2 over 15m)             	volcano                                 	0/0 tasks in gang unschedulable: pod group is not ready, 1 minAvailable.
Normal         	Scheduled                               	15m (x3 over 15m)             	volcano                                 	pod group is ready
Warning        	Unschedulable                           	15m (x2 over 15m)             	volcano                                 	0/0 tasks in gang unschedulable: pod group is not ready, 1 minAvailable.
Normal         	Scheduled                               	15m (x4 over 15m)             	volcano                                 	pod group is ready
```

**Environment**:
- Volcano Version: Volcano is from  `kubectl apply -f https://raw.githubusercontent.com/volcano-sh/volcano/v1.3.0/installer/volcano-development.yaml` cmd
- Kubernetes version (use `kubectl version`): 1.21

Note: The job fails as expected if I remove the retry policy section: 

```
    policies:
    - action: RestartJob
      event: PodEvicted
    - action: RestartJob
      event: PodFailed
```

**Comment by d4l3k** [comment]
is this a duplicate of https://github.com/volcano-sh/volcano/issues/1651 which is already fixed?

**Comment by shinytang6** [comment]
yeah, seems to be fixed by https://github.com/volcano-sh/volcano/pull/1657, but not included in version 1.3.0

**Comment by aivanou** [comment]
Thank you for the fast reply!
